provider/datagen/src/bin/datagen.rs

```diff
@@ -2,13 +2,194 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
-use clap::{crate_authors, crate_version, Command, Arg, ArgGroup};
+use clap::{crate_authors, crate_version, Arg, ArgGroup, Command, Parser, ValueEnum};
 use eyre::WrapErr;
 use icu_datagen::prelude::*;
 use simple_logger::SimpleLogger;
 use std::path::{Path, PathBuf};
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+enum Format {
+    Dir,
+    Blob,
+    Mod,
+    DeprecatedDefault,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+enum Syntax {
+    Json,
+    Bincode,
+    Postcard,
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+enum TrieType {
+    Small,
+    Fast,
+}
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, ValueEnum)]
+enum CliCollationHanDatabase {
```

Comment:
Suggestion: perhaps these enums should go in a `mod cli { ... }` since they seem likely to collide with other things. `Syntax`, `TrieType`, `Format`, etc., all exist elsewhere throughout ICU4X